tools/tslint-rules/coercionTypesRule.ts

```diff
@@ -140,19 +140,22 @@ class Walker extends Lint.RuleWalker {
     const directiveDecorator =
         node.decorators.find(decorator => isDecoratorCalled(decorator, 'Directive'));
 
-    // We should lint non-abstract directives.
     if (directiveDecorator) {
-      const args = (directiveDecorator.expression as ts.CallExpression).arguments;
-      const directiveConfig = args.length && ts.isObjectLiteralExpression(args[0]) ?
-          args[0] as ts.ObjectLiteralExpression : null;
-      const selector = directiveConfig ? directiveConfig.properties.find(prop => {
-        return prop.name && ts.isIdentifier(prop.name) && prop.name.text === 'selector';
-      }) : null;
-
-      return !!selector && ts.isPropertyAssignment(selector) &&
-        (ts.isStringLiteral(selector.initializer) ||
-         ts.isNoSubstitutionTemplateLiteral(selector.initializer)) &&
-         !selector.initializer.text.startsWith('do-not-use-abstract-');
+      const firstArg = (directiveDecorator.expression as ts.CallExpression).arguments[0];
+      const metadata = firstArg && ts.isObjectLiteralExpression(firstArg) ? firstArg : null;
+      const selectorProp = metadata ?
+          metadata.properties.find((prop): prop is ts.PropertyAssignment => {
+            return ts.isPropertyAssignment(prop) && prop.name && ts.isIdentifier(prop.name) &&
+                prop.name.text === 'selector';
+          }) :
+          null;
+      const selectorText =
+          selectorProp != null && ts.isStringLiteralLike(selectorProp.initializer) ?
```

Comment:
We probably don't need to look at the selector's text, just that the property is defined. I think the compiler will throw if the selector is invalid as well.

Reply:
One could still set the selector to `undefined`, so I thought I'll cover that.